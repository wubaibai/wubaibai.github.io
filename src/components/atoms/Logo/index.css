@keyframes swing-a {
	0% {
		transform: translateX(0);
	}

	25% {
		transform: translateX(-4px);
	}

	50% {
		transform: translateX(0);
	}

	75% {
		transform: translateX(3px);
	}

	100% {
		transform: translateX(0);
	}
}

@keyframes swing-b {
	0% {
		transform: translateX(0);
	}

	25% {
		transform: translateX(-2px);
	}

	50% {
		transform: translateX(0);
	}

	75% {
		transform: translateX(2px);
	}

	100% {
		transform: translateX(0);
	}
}

.up {
	background-color: var(--secondary);
	z-index: 2;
}

.down {
	background-color: var(--lightBlue);
	z-index: 1;
	margin-left: 25px;
	margin-top: -20px;
}

.logo {
	padding-right: 20px;
	margin-right: 40px;

	& > div {
		display: block;
		width: 64px;
		height: 34px;
		border-radius: 100px;
		position: relative;
	}

	&.animate {
		& > div {
			animation-duration: 1.6s;
			animation-iteration-count: infinite;
			animation-timing-function: ease-in;
		}

		& > .up {
			animation-name: swing-a;
		}

		& > .down {
			animation-name: swing-b;
		}
	}
}
