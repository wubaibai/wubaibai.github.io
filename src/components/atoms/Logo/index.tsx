import React from 'react';
import classNames from 'classnames';

import style from './index.css';

interface LogoProperty {
	animate?: boolean;
	className?: string;
}

const Logo: React.FC<LogoProperty> = ({ animate, className }) => (
	<div className={classNames(style.logo, animate ? style.animate : undefined, className)}>
		<div className={style.up} />
		<div className={style.down} />
	</div>
);

export default Logo;
