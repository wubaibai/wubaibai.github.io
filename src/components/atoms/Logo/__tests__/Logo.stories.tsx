import React from 'react';
import { storiesOf } from '@storybook/react';

import Logo from 'components/atoms/Logo';

const stories = storiesOf('atoms/Logo', module);

stories.add('__interactive', () => <Logo />, { jest: 'Logo' });
