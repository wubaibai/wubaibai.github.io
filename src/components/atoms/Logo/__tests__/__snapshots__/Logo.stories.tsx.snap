// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__animate 1`] = `
<DocumentFragment>
  <div
    class="logo animate"
  >
    <div
      class="up"
    />
    <div
      class="down"
    />
  </div>
</DocumentFragment>
`;

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="logo"
  >
    <div
      class="up"
    />
    <div
      class="down"
    />
  </div>
</DocumentFragment>
`;
